### PR TITLE
Added Proptypes

### PR DESCRIPTION
**Pull request by kennethlien**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

* Missing `prop-types` and type errors have been corrected.
* Updated to lint on commit instead of on save. To bypass the linter, run `git commit --no-verify`

In general, run `yarn run lint-all` to do a sweep, and click around on the code to see if any yellow things pop up.

### Converted stateless components to functional components
- `CenterLocation` 
- `RewardsCard`
- `Hamburger`
- `CircleIcon`
- `ResourceCard`
- `ResourceCategoryBar`

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: components/product/StoreProducts.js [hunk 1]
```diff
@@ -28,7 +24,10 @@ function StoreProducts({ navigation, store, products }) {
                 store,
               })
             }>
-            <Subhead color="black">See all {products.length}</Subhead>
+            <Subhead color="black">
+              See all
```
Comment (anniero98) [inline components/product/StoreProducts.js:28]: Nit: You'll have to wrap this with {` `} for the linter

File: components/rewards/PointsHistory.js [hunk 2]
```diff
@@ -32,4 +33,11 @@ function PointsHistory({ transactions, user, updates, navigation }) {
   return <View />;
 }
 
+PointsHistory.propTypes = {
+  transactions: PropTypes.object.isRequired,
+  user: PropTypes.object.isRequired,
```
Comment (anniero98) [inline components/rewards/PointsHistory.js:38]: This gets updated in #48 (sorry), but the linter should show you that only `transactions` is actually being referenced (the others are defined and never used), so you don't need `user`, `updates`, or `navigation`

File: components/rewards/RewardsHome.js [hunk 3]
```diff
@@ -14,8 +15,8 @@ import RewardsCard from './RewardsCard';
  * */
 
 function createList(N) {
-  let list = [];
-  for (var i = 1; i <= N; i++) {
+  const list = [];
```
Comment (anniero98) [inline components/rewards/RewardsHome.js:18]: Fixed in #48 too

File: components/rewards/TabBarIcon.js [hunk 4]
```diff
@@ -12,3 +13,12 @@ export default function TabBarIcon(props) {
     />
   );
 }
+
+RewardsHome.propTypes = {
```
Comment (anniero98) [inline components/rewards/TabBarIcon.js:17]: This looks like an incorrect c/p

File: components/product/ProductInfo.js [hunk 5]
```diff
@@ -24,5 +24,3 @@ function ProductInfo({ product }) {
     </ProductInfoContainer>
   );
 }
-
-export default ProductInfo;
```
Comment (anniero98) [inline components/product/ProductInfo.js:28]: I think this is a bug? Please add back

File: components/news/NewsItem.js [hunk 6]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable react/forbid-prop-types */
```
Comment (anniero98) [inline components/news/NewsItem.js:1]: Nit: Remove this line, we changed the rule in `eslintrc.js` right?

File: components/rewards/PointsHistory.js (hunk 2 above)
Comment (anniero98) [reply components/rewards/PointsHistory.js:62]: @kennethlien remember to remove the (now-missing) variables

File: components/rewards/TabBarIcon.js (hunk 4 above)
Comment (anniero98) [reply components/rewards/TabBarIcon.js:17]: bump @kennethlien
